core/src/contract_env.rs

```diff
@@ -107,3 +101,43 @@ impl ContractEnv {
         backend.emit_event(&event.to_bytes().unwrap().into())
     }
 }
+
+pub struct ExecutionEnv {
```

Comment:
Implementing the below functions in `ContractEnv` makes them available for a user writing a module. I think these functions should be hidden somehow. In an ideal world, these functions are called only in generated code.
E.g
```rust
 #[inline]
  pub fn execute_approve(env: odra::ContractEnv) {
      let env_rc = Rc::new(env);
      let exec_env = odra::ExecutionEnv::new(env_rc.clone());
      exec_env.non_reentrant_before();
      let to = exec_env.get_named_arg("to");
      let amount = exec_env.get_named_arg("amount");
      let mut contract = Erc20::new(env_rc);
      let result = contract.approve(to, amount);
      exec_env.non_reentrant_after();
      return result;
  }
```

Otherwise, writting a contract, autocompletion tool would propose these functions, but they are irrelevant in the contract context, or even may result in an unexpected side effect.